AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -349,17 +345,108 @@ - (void)createOrReplaceWithPartition:(NSString *)partition
                      NSTimeInterval interval = [(NSString *)json[kMSDocumentTimestampKey] doubleValue];
                      NSDate *date = [NSDate dateWithTimeIntervalSince1970:interval];
                      NSString *eTag = json[kMSDocumentEtagKey];
-                     MSDocumentWrapper *docWrapper = [[MSDocumentWrapper alloc] initWithDeserializedValue:deserializedDocument
-                                                                                                partition:partition
-                                                                                               documentId:documentId
-                                                                                                     eTag:eTag
-                                                                                          lastUpdatedDate:date];
+                     MSDocumentWrapper *docWrapper = [[MSDocumentWrapper alloc]
+                         initWithDeserializedValue:deserializedDocument
+                                         jsonValue:[[NSString alloc] initWithData:(NSData *)data encoding:NSUTF8StringEncoding]
+                                         partition:partition
+                                        documentId:documentId
+                                              eTag:eTag
+                                   lastUpdatedDate:date];
                      MSLogDebug([MSDataStore logTag], @"Document created/replaced with ID: %@", documentId);
                      completionHandler(docWrapper);
                      return;
                    }];
 }
 
+- (void)listWithPartition:(NSString *)partition
+             documentType:(Class)documentType
+              readOptions:(MSReadOptions *_Nullable)readOptions
+        continuationToken:(nullable NSString *)continuationToken
+        completionHandler:(MSPaginatedDocumentsCompletionHandler)completionHandler {
+  NSMutableDictionary *additionalHeaders = [NSMutableDictionary new];
+  if (continuationToken) {
+    [additionalHeaders setObject:(NSString *)continuationToken forKey:kMSDocumentContinuationTokenHeaderKey];
+  }
+  // Call cosmos DB.
+  [self performOperationForPartition:partition
+                          documentId:nil
+                          httpMethod:kMSHttpMethodGet
+                                body:nil
+                   additionalHeaders:additionalHeaders
+                   completionHandler:^(NSData *_Nullable data, NSHTTPURLResponse *_Nullable response, NSError *_Nullable cosmosDbError) {
+                     // If not OK.
+                     if (!data || [MSDataSourceError errorCodeFromError:cosmosDbError] != MSACDocumentSucceededErrorCode) {
+                       MSLogError([MSDataStore logTag], @"Not able to retrieve documents: %@", [cosmosDbError description]);
```

Comment:
We use `localizedDescription` for logs but I think this is correct. We shouldn't use localized description for logs.

Reply:
About logs, I want us to normalize logging in the "fit and finish" feature to make sure we do the same thing between Apple and Android platforms. @clpolet Also mentioned we probably should not log documents payload (would sensitive information be there), I think that's fair and would be part of this normalization.

Reply:
After looking into `NSError` and `NSObject` implementation, I should recall my previous comment. I got confused of *localized*.
`description` is in `NSObject` and it doesn't give you any of error details. You have to use `localizedDescription` which is a property of `NSError`. Sorry for wrong info.
```suggestion
                       MSLogError([MSDataStore logTag], @"Not able to retrieve documents: %@", [cosmosDbError localizedDescription]);
```